src/imagescache.cpp

```diff
@@ -0,0 +1,291 @@
+            info_.pack = title_;
+            info_.name = emoji.getAttribute(QStringLiteral("title")).trimmed();
+            QString url_ = aclass.getAttribute(QStringLiteral("href")).trimmed();
+            if (!url_.startsWith(QStringLiteral("http"))) {
+                url_.prepend("https://slackmojis.com");
+            }
+            info_.url = QUrl(url_);
+            info_.fileName = aclass.getAttribute(QStringLiteral("download")).trimmed();
+            info_.cached = false; //to check later
+            _images[info_.name] = info_;
+            //qDebug() << "adding emoji:" << info_.pack << info_.name << info_.url << info_.fileName;
+        }
+
+    }
+    return true;
+}
+
+bool ImagesCache::parseEmojiCheactSheet(const QByteArray &data)
+{
+    auto doc = QGumboDocument::parse(data.data());
+    auto root = doc.rootNode();
+    auto nodes = root.getElementsByTagName(HtmlTag::UL);
+    //qDebug() << "emojis" << nodes.size();
+
+    for (const auto& node: nodes) {
+        const QStringList & list_ = node.classList();
+        if (list_.isEmpty()) {
+            continue;
+        }
+        QString title_ = node.classList().at(0).trimmed();
+        //qDebug() << "title" << title_;
+        auto emojis = node.getElementsByClassName(QStringLiteral("emoji"));
+        auto emojisnames = node.getElementsByClassName(QStringLiteral("name"));
+        if (emojis.size() != emojisnames.size()) {
+            qDebug() << "emojis parse something wrong" << emojis.size() << emojisnames.size();
+            continue;
+        }
+        for (ulong i = 0; i < emojis.size(); i++) {
+            const auto& emoji = emojis.at(i);
+            const auto& name = emojisnames.at(i);
+
+            ImageInfo info_;
+            info_.pack = title_;
+            info_.name = name.innerText().trimmed();
+            QString url_ = emoji.getAttribute(QStringLiteral("data-src")).trimmed();
+            if (!url_.startsWith(QStringLiteral("http"))) {
+                url_.prepend("https://www.webpagefx.com/tools/emoji-cheat-sheet/");
+            }
+            info_.url = QUrl(url_);
+            info_.fileName = info_.url.fileName();
+            info_.cached = false; //to check later
+            _images[info_.name] = info_;
+            //qDebug() << "adding emoji:" << info_.pack << info_.name << info_.url << info_.fileName;
+        }
+    }
+    return (_images.size() > 0);
+}
+
+void ImagesCache::requestSlackMojis()
+{
+    QNetworkReply *reply = _qnam.get(QNetworkRequest(QUrl(QStringLiteral("https://slackmojis.com/"))));
+    connect(reply, &QNetworkReply::finished, this, &ImagesCache::onImagesListRequestFinished);
+}
+
+void ImagesCache::requestEmojiCheactSheet()
+{
+    QNetworkReply *reply = _qnam.get(QNetworkRequest(QUrl(QStringLiteral("https://www.webpagefx.com/tools/emoji-cheat-sheet/"))));
+    connect(reply, &QNetworkReply::finished, this, &ImagesCache::onImagesListRequestFinished);
+}
+
+void ImagesCache::parseJson(const QByteArray &data) {
```

Comment:
could probably be a lot simpler with just using QDataStream, but doesn't matter much